### DISCUSSION TITLE
New technique to combine quality and speed: ThinSkin. Thin layers on the outside and thicker layers on the inside.

### DISCUSSION THREAD

**Discussion by john3volts3** [smartavionics/Cura#224] [Ideas]
To achieve a nice exterior finish, the use of thinner layers is very effective. But it slows down the printing process. So why print all the walls of the object with this thickness of layers, when it doesn't add anything to the mass because it doesn't show? My idea is very simple: we print the outer walls in thin layers and the inner walls in thicker layers. And this is quite possible. All the programmers have to do is implement this function in their slicers. In the meantime, I have been experimenting with the POC on a very simple object. All details are available on printable : https://www.printables.com/model/482817-thinskin-speed-and-quality-combined-one-more-trick

![thinSkin-2](https://github.com/smartavionics/Cura/assets/101920483/641ce5c7-4f05-4932-abfe-682db5df0e96)

https://github.com/smartavionics/Cura/assets/101920483/9c85d2e8-fb3b-4ac8-bb71-4f94fb398913

https://github.com/smartavionics/Cura/assets/101920483/daa070fa-231f-4117-8318-b297a53ace59

https://github.com/smartavionics/Cura/assets/101920483/cc1d3758-f152-4107-8711-c383816fdbcd

https://github.com/smartavionics/Cura/assets/101920483/0c07c622-b597-40f4-9d9f-32dd44088f8d

![layers](https://github.com/smartavionics/Cura/assets/101920483/dc7c669b-cb28-4636-a108-249584d7ba97)
![microscope-9](https://github.com/smartavionics/Cura/assets/101920483/2445181d-97e0-4207-bd9b-5a0a455dd9e2)
![microscope-12](https://github.com/smartavionics/Cura/assets/101920483/69385de1-5a8b-4352-8821-2722d733b50a)

https://youtu.be/oHky-r_hI2c

**Comment by Sophist-UK** [comment]
I seem to remember that Cura has the ability to print double thickness infill layers already!!

But I don't think it can print thicker bottom surfaces and thin walls, or thin outer walls (with Outer Walls first) and thicker inner walls.

**Reply by john3volts3** [reply]
Hello, As far as I know, Cura cannot combine walls of different thicknesses on the same layer. 
Here I make a suggestion for a new feature to add to slicers, because it would save time when you want a better finish on the outside.

**Reply by 5axes** [reply]
> I seem to remember that Cura has the ability to print double thickness infill layers already!!

I would like to take this opportunity to point out a difference between versions 4.20 and the standard branch of Cura. In Cura it is indeed possible to define a different layer height for the internal infill (**Infill Layer Thickness**). If we could use the **_Extra Infill Wall Count_** parameter at the same time we would have the result asked for.
But with the Standard version, the inner walls have the same thickness as the outer walls.
![image](https://github.com/smartavionics/Cura/assets/11015345/d3a1e17b-6796-4cd9-a298-a813b926a0c8)
and in the version of @smartavionics the walls are simply not generated in this case, Bug ou Feature ?
![image](https://github.com/smartavionics/Cura/assets/11015345/82e29ddd-de75-41e9-9d64-bf732ecd38fa)

**Comment by smartavionics** [comment]
> and in the version of @smartavionics the walls are simply not generated in this case, Bug ou Feature ?

Yes, that's a bug but I think it was in the 4.13 code that my variant is based on. It should be easy enough to make it work for a future release.

**Comment by 5axes** [comment]
Ok so solved in 4.20.13

Define :
 Layer Height : 0.1
 Line Width : 0.4
 Wall Line Count  : 2

 Infill Layer Width : 0.4
 Infill Layer Thickness : 0.3
 Extra Infill Wall Line Count  : 4

![image](https://github.com/smartavionics/Cura/assets/11015345/397cc236-9c28-44cd-8ba2-1420db815ba6)

Need to be tested ... It's something for @CNCKitchen :)

**Reply by 5axes** [reply]
Well, the result isn't exactly what I'd hoped for, and at the moment I'm having a bit of trouble explaining what I'm seeing.
If I calculate the theoretical flow, this is what we should get: 

Wall | Height | Width | Speed | Flow
-- | -- | -- | -- | --
Infill | 0,3 mm | 0,4 mm | 60 mm/s | 7,2 mm³/s
Inner   Wall | 0,1 mm | 0,4 mm | 40 mm/s | 1,6 mm³/s
Outer   Wall | 0,1 mm | 0,4 mm | 30 mm/s | 1,2 mm³/s


and that's what I have in Cura  : 
![image](https://github.com/smartavionics/Cura/assets/11015345/80a9e864-ecb3-4f58-83a3-eee8cdf0326e)

![image](https://github.com/smartavionics/Cura/assets/11015345/b44e5f3b-d7d0-4cdd-ab24-9f86d0a0599e)

![image](https://github.com/smartavionics/Cura/assets/11015345/7c58782d-5b85-4282-823b-a61521989ff8)


But, in the Gcode simulation, the infill is clearly under-extruded:  
![image](https://github.com/smartavionics/Cura/assets/11015345/e9266e3a-d62d-435b-a308-c7c93ae1fb09)

And on the printed part I find this under-extrusion.
![image](https://github.com/smartavionics/Cura/assets/11015345/9ae4dac0-897a-419d-8f14-f817997d5f63)

To get a correct result I have to increase the filling flow to 200%.
![image](https://github.com/smartavionics/Cura/assets/11015345/55492828-d7e6-4418-bebf-352f6a97e3fb)

![image](https://github.com/smartavionics/Cura/assets/11015345/737c813f-f082-4b4a-baba-712edd70c761)

And even with this settings the printed part is still slightly under-extruded???
![image](https://github.com/smartavionics/Cura/assets/11015345/cad497ac-161a-456d-a260-a324385e187c)

**Reply by 5axes** [reply]
I ve analysed the Gcode , it's correct the E extrusion values are correct , the requested flow is ok (7,2 or 14,4 if the flow is set to 200%) 
If it was just a question of result on the printer, I would say my extruder is not able to generate this flow (nozzle size 0.4 mm) but as we have also a visualisation in the Gcode simulation of this result, I am clearly on something that I cannot explain . Any thought ?

@GregValiant as you also use an Ender3 do you have any explanation ?

**Comment by smartavionics** [comment]
Under-extrusion due to too high a flow rate is extremely common. I think that people believe that their extruder will work just fine up to some limit where it will suddenly not be able to deliver the expected rate. But in reality, the achieved flow rate will start reducing much earlier, it may not be a linear function but the achieved flow rate will taper off as the demanded flow rate increases.

One of my printers uses a Duet (aka RepRap) controller and it has a configuration option whereby you can specify a transformation equation of the form F = a*f^2 + b*f + c where f is the demanded flow and F is the flow actually used. I.e. it is non-linear. The effect of this transformation is to make the extruder more linear so when the demanded flow rate is doubled, say, the output really is doubled (or at least closer to being doubled). Of course, the extruder will run out of steam at some point so you can't actually achieve a better flow performance than what the extruder is physically capable of but this feature helps to linearise the actual flow achieved.

**Comment by GregValiant** [comment]
Not really anything helpful.  We might be talking apples and oranges though.
Some background:  I often run "Infill Layer Thickness" and "Support Infill Layer Thickness" at 2X layer height.  It is extremely rare for me to use anything other than "Grid" infill.
I wrote an Excel macro to use as my analysis tool.  I haven't had to change it in quite a while.  I believe it gives accurate and relevant results.

- @5axes said _"the requested flow is ok (7,2 or 14,4 if the flow is _**set**_ to 200%)"._
-    What I find is that both forks of Cura handle that themselves.  I don't find a need to "set" the Infill Flow to 200%.  I have _not_ noticed any under-extrusion of the walls.
- My printer is a Creality Ender 3 Pro with 1.1.5 8-bit mainboard and a generic all-metal hot end.
- I can print a large skin at 175mm/sec and it will be fine.  Once I get above that speed the extrusions don't start correctly but the extruder will keep up once it gets going.  I never took it further (ran out of interest) but my thought was that adjusting the Acceleration Limit of the E might have allowed me to get good print quality at an even higher print speed.  At any rate, at 175mm/sec and with the settings I used it came out to a 14mm³/sec flow-rate.  I consider that the "Maximum Sustainable Flow-Rate" on my Ender 3 Pro.  Consequently, I can not run 2X Infill Layer Thickness at that speed.  I occasionally print some things at 100mm/sec but my default fast print speed is 75mm/sec.

Apples:
SA 4.20.13 treats "Extra Infill Walls" as Infill.  If you set the "Infill Layer Thickness" to 2X layer height then ALL of the "Infill" will be at 2X layer height and so the Infill Flow Rate will be double (as @5axes numbers "7,2 or 14,4" show and I get the same ratio).
These are in mm³/sec at 75mm/sec print speed, 0.2 LayerHeight and 0.4 LineWidth and are extrapolated from the gcode:
Within both ";TYPE:WALL-OUTER" and ";TYPE:WALL-INNER"
Walls Flow Rate:  6.000
Within ";TYPE:FILL"
Infill - Fill Flow-Rate:  12.000
Infill - Walls Flow-Rate:  **12.000**
A side note is that adjusting "Maximum Speed for Flow Equalization" didn't seem to make any difference.  The print speed (and the Flow-Rate) remained the same regardless of the value I entered.

Oranges:
But UM 5.3.1 treats "Extra Infill Walls" as walls.  When Infill Layer Thickness" is 2X layer height only the interior infill pattern is affected whereas those "extra infill walls" will be at 1X Layer Height just like the Inner and Outer walls.  So looking at the numbers in a flow analysis of a UM Cura gcode there will be different Flow-Rates within the infill section.
Within both ";TYPE:WALL-OUTER" and ";TYPE:WALL-INNER"
Walls Flow Rate:  6.000
Within ";TYPE:FILL"
Infill - Fill Flow-Rate:  12.000
Infill - Walls Flow-Rate:  **6.000**
The Flow Equalization Factor also doesn't appear to make any difference.  But I have seen it make a difference on other models.

@john3volts3 has in interesting concept.  One problem I foresee is that I consider 4 skins to be a minimum to hide the infill pattern.  It's dependent on things like filament color, but the "4 top skin minimum" has been a good general number for me.
Doubling the Skin Layer Height would either cut down the number of skins to 2, or if adjusted the other way, double the number of wall layers.  I can see that being a show-stopper trying to do this in post-process.  It would be necessary to delete the skins from every other layer while doubling the flow of the skins that remain.  Very clumsy unless additional rules are applied (like no odd numbers of skins).

This is from 4.20.13 at a 0.2 Layer Height and the Infill Layer Thickness at 3X Layer Height.  I just thought it was interesting as the "catch up" layers directly below the first top skin are correctly dropped to "Layer Height" to eliminate any gap.  The extra flow appears to affect the displayed line width where it should really affect the displayed layer height as without any squish the extrusion won't spread.
![image](https://github.com/smartavionics/Cura/assets/64202104/826a7d93-9055-4f17-a9c3-255e363c7248)

Time for another cup of coffee.
